### PR TITLE
chore: Emit performance marks for button and table

### PR DESCRIPTION
**Pull request by connorlanigan**
### Description

This change adds performance marks to the Button and Table component. These marks are created when the component is first rendered, and when properties of the component (e.g. `loading` or `disabled`) change.

Related links, issue #, if available: n/a

### How has this been tested?

Added integration tests. I did not add unit tests, since JSDOM does not support the necessary performance APIs.

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pages/table/performance-marks.page.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useState } from 'react';
+import Box from '~components/box';
+import { Header, Link, SpaceBetween, Table, Tabs } from '~components';
+
+export default function ButtonsScenario() {
```
Comment (michaeldowseza) [inline pages/table/performance-marks.page.tsx:7]: Scenario name

File: pages/button/performance-marks.page.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        <label>
+          <input type="checkbox" checked={disabled} onChange={e => setDisabled(e.target.checked)} id="disabled" />
+          Disabled
+        </label>
+        <Button variant="primary" loading={loading} disabled={disabled}>
+          Primary button
+        </Button>
+        <Button>Non-primary button</Button>
+
+        <Tabs
```
Comment (michaeldowseza) [inline pages/button/performance-marks.page.tsx:28]: Would Modal be a good test case as the content is "rendered" but is invisible

File: src/internal/hooks/use-performance-marks.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (michaeldowseza) [inline src/internal/hooks/use-performance-marks.ts:1]: For now its probably fine to have this here. I imagine it'll move to component-toolkit eventually once we want to share with the board components package.

File: src/button/internal.tsx [hunk 4]
```diff
@@ -80,6 +81,18 @@ export const InternalButton = React.forwardRef(
     const { stepNumber, stepNameSelector } = useFunnelStep();
     const { subStepSelector, subStepNameSelector } = useFunnelSubStep();
 
+    usePerformanceMarks(
+      'primaryButton',
```
Comment (michaeldowseza) [inline src/button/internal.tsx:85]: Why "primaryButton" as opposed to "Button"?

Comment (connorlanigan) [reply src/button/internal.tsx:85]: The alternative I see is using `button` and including the variant value in the mark details, but that might create the expectation that the mark is also emitted for non-primary buttons. If/when we expand the performance marks to more components/variants, I'd replace it with `button`.

File: src/internal/hooks/use-performance-marks.ts (hunk 3 above)
Comment (connorlanigan) [reply src/internal/hooks/use-performance-marks.ts:1]: I agree, we'll probably have to move it in the future.

File: pages/button/performance-marks.page.tsx (hunk 2 above)
Comment (connorlanigan) [reply pages/button/performance-marks.page.tsx:30]: Good idea! Added in the new revision

**Review by michaeldowseza — APPROVED** [review]
LGTM!

**Pull request merged**